modules/apps/document-library/document-library-api/src/main/java/com/liferay/document/library/versioning/VersioningPolicy.java

```diff
@@ -25,6 +25,7 @@
 public interface VersioningPolicy {
 
 	public Optional<DLVersionNumberIncrease> computeDLVersionNumberIncrease(
-		DLFileVersion previousDLFileVersion, DLFileVersion nextDLFileVersion);
+		DLFileVersion previousDLFileVersion, DLFileVersion nextDLFileVersion,
+		String[] newTagNames);
```

Comment:
@davidrison I didn't like the solution of passing the tags here. I'd rather pass the ServiceContext object but Source Formatter didn't allow me do do it:

Do not use type 'ServiceContext' in API method signature, see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/documentation/api_method_signatures.markdown: ./src/main/java/com/liferay/document/library/versioning/VersioningPolicy.java (SourceCheck:JavaAPISignatureCheck)

Reply:
Normally, that would be true but we use the service context in other areas so it would be ok to do this.  For now we can leave it this way but decide later if it makes sense to change to using service context.